### PR TITLE
Bug/lad 142/whiteboard performance part1

### PR DESCRIPTION
**Pull request by AlexSapple**
expose configuration that (initially) allows the following save frequency policies;

* ALL - save as per today (this triggers save events for both "version-broadcast" scenarios and on user exit
    * note that version broadcasts are not the same as a save event, they are the tigger that effectively informs the media server that something has changed - the media server then responds with it's own emit to inform the app to save based on it's logic of when a save is allowed - this covers the periodic in meeting saves
*ONEXIT - save as users exit, this triggers a save event on any user who's socket is disconnecting
*NONE - simply do not trigger a save event (regardless of the onSaveNeeded listener) 

* bonus bug - the version-broadcast was firing on !volatile **||** scene_update - this was incorrect and a result of a previous change - this should have been !volatile **&&** scene_update

### CODE REVIEW

**Review by laduma-adamc — APPROVED** [review]

File: src/laduma-api/api.ts [hunk 1]
```diff
@@ -64,10 +70,14 @@ class LadumaApi {
...
   }
 
-  public OnSaveNeeded(eventArgs: SaveBoardEventArgs)
-  {
-    if(this._onSaveNeeded)
+  public getSaveFrequency(): SaveFrequency {
+    return this._saveFrequency;
+  }
+
+  public OnSaveNeeded(eventArgs: SaveBoardEventArgs) {
```
Comment (carboni-rob) [inline src/laduma-api/api.ts:77]: Nitpick of the day:
```suggestion
  public onSaveNeeded(eventArgs: SaveBoardEventArgs) {
```

File: src/types.ts [hunk 2]
```diff
@@ -158,13 +158,15 @@ export type ExcalidrawAPIRefValue =
...
-  snapshot: any,
-  state: any
+  elements: any;
+  snapshot: any;
+  state: any;
 };
 
+/** The frequency to fire the onSaveNeededEvent (if applicable) */
+export type SaveFrequency = "ALL" | "ONEXIT" | "NONE";
+
```
Comment (carboni-rob) [inline src/types.ts:169]: Here we can also do:
```suggestion
/** The frequency to fire the onSaveNeededEvent (if applicable) */
export enum SaveFrequency {
  All= 'ALL',
  OnExit='ONEXIT',
  None= 'NONE'
}
```
And use as
```
        ladumaApi.getSaveFrequency() === SaveFrequency.All
```
which has the advantage of leveraging TS' typeahead. Also, if in the future we have to change one of the enum's elements for some reason, will save us from changing in multiple locations.

**Review by carboni-rob — APPROVED** [review]
LGTM! 🚀

File: src/types.ts (hunk 2 above)
Comment (AlexSapple) [reply src/types.ts:169]: thanks @carboni-rob , I thought about using enum (and in fact I did start using it this way), but then I went in the other direction for no specific reason! :) . Are you happy if I raise this and the name correction above as a small separate code tidy up task?

**Pull request merged**